src/components/Navbar.tsx

```diff
@@ -0,0 +1,142 @@
+import { GoHome } from "react-icons/go";
+import { PiCalendarCheckFill } from "react-icons/pi";
+import { TiGroup } from "react-icons/ti";
+import { PiIdentificationBadgeFill } from "react-icons/pi";
+import { FiLogOut } from "react-icons/fi";
+import { useState, useEffect } from "react";
+import { useAuth } from "@providers";
+import Hamburger from "hamburger-react";
+
+const navItems = [
+    { path: "/profile", label: "Home", Icon: GoHome },
+    { path: "/schedule", label: "Schedule", Icon: PiCalendarCheckFill },
+    { path: "/networking", label: "Networking", Icon: TiGroup },
+    { path: "/ticket", label: "Ticket", Icon: PiIdentificationBadgeFill },
+];
+
+export const Navbar = () => {
+    const { logout } = useAuth();
+
+    const [isMobile, setIsMobile] = useState(window.innerWidth <= 768);
+    const [isMobileMenuOpen, setMobileMenuOpen] = useState(false);
+
+    const updateNavbarState = () => {
+        setIsMobile(window.innerWidth <= 768);
+    };
+
+    useEffect(() => {
+        window.addEventListener("resize", updateNavbarState);
+        return () => {
+            window.removeEventListener("resize", updateNavbarState);
+        };
+    }, []);
+
+    const renderNavItems = (isMobile: boolean) =>
+        navItems.map(({ path, label, Icon }) => (
+            <li
+                key={label}
+                className="p-4 hover:bg-slate-100 duration-300 transition-colors rounded-md w-full hover:text-black"
+            >
+                <a
```

Comment:
The `<a`> tags need to be replace with the `<Link>` component from `react-router-dom`. The native tag causes the page to do a reload, while the `<Link>` does a repaint with the react app that has already been downloaded.

Reply:
resolved in newest commit